dartagnan/src/main/java/com/dat3m/dartagnan/verification/model/event/DefaultEventModel.java

```diff
@@ -0,0 +1,99 @@
+package com.dat3m.dartagnan.verification.model.event;
+
+import com.dat3m.dartagnan.program.event.core.*;
+import com.dat3m.dartagnan.program.event.Event;
+import com.dat3m.dartagnan.program.event.RegReader;
+import com.dat3m.dartagnan.program.event.RegWriter;
+import com.dat3m.dartagnan.verification.model.ThreadModel;
+
+import static com.dat3m.dartagnan.program.event.Tag.*;
+
+
+public class DefaultEventModel implements EventModel {
+    protected final Event event;
+    protected final ThreadModel thread;
+    protected int id = -1;
+
+    public DefaultEventModel(Event event, ThreadModel thread, int id) {
+        this.event = event;
+        this.thread = thread;
+        this.id = id;
+
+        thread.addEvent(this);
+    }
+
+    @Override
+    public Event getEvent() {
+        return event;
+    }
+
+    @Override
+    public ThreadModel getThreadModel() {
+        return thread;
+    }
+
+    @Override
+    public int getId() {
+        return id;
+    }
+
+    @Override
+    public boolean isMemoryEvent() { return event.hasTag(MEMORY); }
+
+    @Override
+    public boolean isInit() { return event.hasTag(INIT); }
+
+    @Override
+    public boolean isWrite() { return event.hasTag(WRITE); }
+
+    @Override
+    public boolean isRead() { return event.hasTag(READ); }
+
+    @Override
+    public boolean isFence() { return event.hasTag(FENCE); }
+
+    @Override
+    public boolean isExclusive() { return event.hasTag(EXCL); }
+
+    @Override
+    public boolean isRMW() { return event.hasTag(RMW); }
+
+    @Override
+    public boolean isVisible() { return event.hasTag(VISIBLE); }
+
+    @Override
+    public boolean isJump() { return event instanceof CondJump; }
+
+    @Override
+    public boolean isAssert() { return event instanceof Assert; }
```

Comment:
Are these even used somewhere? I would get rid of all unused one's and maybe even all of them.
Also, it's probably best to replace tag-based checks by instanceof checks if possible (e.g., for read, write, and init).

Reply:
Why should tag-based checks be avoided? For read and write there are so many different event classes, we should check them one by one by instanceof?

Reply:
I don't think there are. For write there is only `Store` and for read there is only `Load` as far as I can see. Classes like `Init` and `RMWStore` also inherit from `Store` and thus are covered by the `instanceof` check.
All non-core classes should not exist anymore at this stage (they have been compiled to core events).
There is a class called `CoreCodeVerification` which checks for all possible classes that can exist.